Source/WebKit/UIProcess/API/Cocoa/NSAttributedString.mm

```diff
@@ -456,11 +456,11 @@ + (void)loadFromHTMLWithData:(NSData *)data options:(NSDictionary<NSAttributedSt
 {
     [self _loadFromHTMLWithOptions:options contentLoader:^WKNavigation *(WKWebView *webView) {
         auto* textEncodingName = dynamic_objc_cast<NSString>(options[NSTextEncodingNameDocumentOption]);
-        auto characterEncoding = static_cast<NSStringEncoding>(dynamic_objc_cast<NSNumber>(options[NSCharacterEncodingDocumentOption]).unsignedIntegerValue);
         auto* baseURL = dynamic_objc_cast<NSURL>(options[NSBaseURLDocumentOption]);
+        unsigned long characterEncoding = dynamic_objc_cast<NSNumber>(options[NSCharacterEncodingDocumentOption]).unsignedLongValue;
 
         if (characterEncoding && !textEncodingName) {
-            auto stringEncoding = CFStringConvertNSStringEncodingToEncoding(characterEncoding);
+            CFStringEncoding stringEncoding = CFStringConvertNSStringEncodingToEncoding(characterEncoding);
```

Comment:
And then CFStringConvertNSStringEncodingToEncoding takes an unsigned long for some reason, but an unsigned int fits in this type, so no change seems to be necessary.

Reply:
The other example of this function being called in WebKit does have the unsigned long explicitly, so that's inconsistent with this. 

But this is about the arguments to the IANAChar function.

Reply:
> But this is about the arguments to the IANAChar function.

CFStringConvertEncodingToIANACharSetName takes CFStringEncoding, which is what we pass with or without this PR. It just changes the variable declaration to not use auto, but the actual type remains the same.

Reply:
Documention says one thing, but the function signature for CFStringConvertNSStringEncodingToEncoding is another. I am following the signature, because that takes an unsigned long explicitly, not an NSUInteger.